### PR TITLE
Map the new SleepManager

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/server/world/SleepManager.mapping [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	METHOD method_33813 isLongEnough (ILjava/util/List;)Z
+		ARG 1 percentage
+		ARG 2 players
+	METHOD method_33814 update (Ljava/util/List;)Z
+		COMMENT Updates the sleeping players.
+		COMMENT
+		COMMENT @return {@code true} if the sleeping players or total players have
+		COMMENT changed
+		ARG 1 players
+	METHOD method_33815 countSleeping ()I
```
Comment (haykam821) [inline mappings/net/minecraft/server/world/SleepManager.mapping]: This just returns `sleeping`:

```suggestion
	METHOD method_33815 getSleeping ()I
```

File: mappings/net/minecraft/server/world/SleepManager.mapping [hunk 2]
```diff
@@ -0,0 +1,18 @@
+CLASS net/minecraft/class_5838 net/minecraft/server/world/SleepManager
+	FIELD field_28866 total I
+	FIELD field_28867 sleeping I
+	METHOD method_33811 clearSleeping ()V
+	METHOD method_33812 canSkipNight (I)Z
+		ARG 1 percentage
+	METHOD method_33813 isLongEnough (ILjava/util/List;)Z
```
Comment (haykam821) [inline mappings/net/minecraft/server/world/SleepManager.mapping]: I would recommend changing this name along with the one in `PlayerEntity` to better indicate that this is based off the player's sleep timer. Would something like `PlayerEntity#hasFullySlept` and `SleepManager#haveAllFullySlept` work?

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/server/world/SleepManager.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/server/world/SleepManager.mapping]: Is there one in player entity? The whole check is done in this method I recall

File: mappings/net/minecraft/server/world/SleepManager.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/server/world/SleepManager.mapping]: sounds weird... it's like a mix of boolean and getter

File: mappings/net/minecraft/server/world/SleepManager.mapping (hunk 2 above)
Comment (haykam821) [reply mappings/net/minecraft/server/world/SleepManager.mapping]:
```java
public boolean isLongEnough(int percentage, List<ServerPlayerEntity> players) {
	int sleepingLongEnoughCount = players.stream().filter(PlayerEntity::isSleepingLongEnough).count();
	return sleepingLongEnoughCount >= this.getNightSkippingRequirement(percentage);
}
```

Comment (liach) [reply mappings/net/minecraft/server/world/SleepManager.mapping]: Maybe `hasSleptLongEnough`. If you make a pr to supersede this one, I will endorse it

File: mappings/net/minecraft/server/world/SleepManager.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/server/world/SleepManager.mapping]: Updated to `getSleeping`.

File: mappings/net/minecraft/server/world/SleepManager.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/server/world/SleepManager.mapping]: Decided to call this `canResetTime` vs the plain `canSkipNight`

**Review by YanisBft — APPROVED** [review]

**Review by enbrain — APPROVED** [review]

**Pull request merged**